### PR TITLE
Resolve GHC warnings

### PR DESCRIPTION
**Pull request by tadfisher**
- `tyVarBndrs` is unused
- `tyVarBndrNoSig` is unused
- ~~`-Wno-incomplete-record-udpates` is spurious, but I don't know how to resolve it for this:~~

```haskell
giveBase :: Maybe MoatType -> [Protocol] -> MoatData -> MoatData
giveBase r ps = \case
  s@MoatStruct {} -> s {structPrivateTypes = [giveProtos ps (suffixBase (stripFields s))]}
  s@MoatEnum {} -> s {enumPrivateTypes = [giveProtos ps (suffixBase (stripFields s)) {enumRawValue = r}]}
                                                     -- ^=============================================$
  s -> s
```

### CODE REVIEW

File: src/Moat.hs [hunk 1]
```diff
@@ -12,6 +12,7 @@
 {-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE ViewPatterns #-}
+{-# OPTIONS_GHC -Wno-incomplete-record-updates #-}
```
Comment (chiroptical) [inline src/Moat.hs]: :thinking:

Comment (chiroptical) [reply src/Moat.hs]: Ahh, I guess this goes along with the incomplete pattern matching thing which is not actually incomplete.

Comment (chiroptical) [reply src/Moat.hs]: I don't think we want this.

Comment (tadfisher) [reply src/Moat.hs]: Will remove.

**Review by chiroptical — APPROVED** [review]
We can work on the other one another time.

**Comment by chiroptical** [comment]
I guess we would just need to ensure the result of `giveProtos` has `enumRawValue`. In theory, it returns what it is given but the compiler doesn't know that.

**Comment by tadfisher** [comment]
> I guess we would just need to ensure the result of `giveProtos` has `enumRawValue`. In theory, it returns what it is given but the compiler doesn't know that.

Maybe split it up into `giveStructProtos` and `giveEnumProtos`?

**Pull request merged**